File: add-two-numbers/Driver.java
class ListNode {
    int val;
    ListNode next;

    ListNode(int x) { 
        val = x;
    }

    @Override
    public String toString() {

        String output = "( ";
        ListNode curr = this;
        while (curr != null) {
            output += curr.val + " ";
            curr = curr.next;
        }
        return output + ")";
    }
}

public class Driver {
    public static void main(String[] args) {

        Solution tna = new Solution();
        ListNode l1 = new ListNode(2);
        ListNode l1_1 = new ListNode(4);
        ListNode l1_2 = new ListNode(3);
        l1.next = l1_1;
        l1_1.next = l1_2;

        ListNode l2 = new ListNode(5);
        ListNode l2_1 = new ListNode(6);
        ListNode l2_2 = new ListNode(4);
        ListNode l2_3 = new ListNode(5);
        l2.next = l2_1;
        l2_1.next = l2_2;
        l2_2.next = l2_3;

        ListNode l3 = new ListNode(9);
        ListNode l3_1 = new ListNode(9);
        l3.next = l3_1;

        ListNode l4 = new ListNode(1);

        System.out.println(l3);
        System.out.println(l4);
        ListNode result = tna.addTwoNumbers(l3, l4);
        System.out.println(result);
    }    
}

File: smallest-subtree-with-all-the-deepest-nodes/driver.go
package main

import "fmt"

type TreeNode struct {
	Val int
	Left *TreeNode
	Right *TreeNode
}

func main() {

	var root TreeNode
	root.Val = 3
	var l1_1 TreeNode
	l1_1.Val = 5
	root.Left = &l1_1
	var l1_2 TreeNode
	l1_2.Val = 1
	root.Right = &l1_2
	var l2_1 TreeNode
	l2_1.Val = 6
	l1_1.Left = &l2_1
	var l2_2 TreeNode
	l2_2.Val = 2
	l1_1.Right = &l2_2
	var l2_3 TreeNode
	l2_3.Val = 0
	l1_2.Left = &l2_3
	var l2_4 TreeNode
	l2_4.Val = 8
	l1_2.Right = &l2_4
	var l3_1 TreeNode
	l3_1.Val = 7
	l2_2.Left = &l3_1
	var l3_2 TreeNode
	l3_2.Val = 4
	l2_2.Right = &l3_2
	var l3_3 TreeNode
	l3_3.Val = 9
	l2_3.Left = &l3_3

	fmt.Println(subtreeWithAllDeepest(&root).Val)
}
